agent/php_internal_instrument.c

```diff
@@ -1527,10 +1560,63 @@ NR_INNER_WRAPPER(mysqli_stmt_prepare) {
  */
 NR_INNER_WRAPPER(memcache_function) {
   nr_php_instrument_datastore_operation_call(nr_wrapper, NR_DATASTORE_MEMCACHE,
-                                             nr_wrapper->extra, NULL,
+                                             nr_wrapper->extra, NULL, false,
                                              INTERNAL_FUNCTION_PARAM_PASSTHRU);
 }
 
+NR_INNER_WRAPPER(memcached_connect_function) {
+  char* host = NULL;
+  nr_string_len_t host_len = 0;
+  zend_long port = 0;
+  zend_long weight = 0;
+  nr_datastore_instance_t* instance = NULL;
+
+  if (SUCCESS
+      == zend_parse_parameters_ex(
+          ZEND_PARSE_PARAMS_QUIET, ZEND_NUM_ARGS() TSRMLS_CC, "s|ll", &host,
+          &host_len, &port, &weight)) {
+    if (NULL != host) {
+      instance = nr_php_memcached_create_datastore_instance(host, port);
```

Comment:
Where does `instance` get freed?

Reply:
It gets freed during nr_php_instrument_datastore_operation_call (or nr_php_instrument_datastore_operation when the underlying call isn't needed).

It is a somewhat confusing code structure, but is replicated from the redis instrumentation

Reply:
This thread is obsolete as of 3ed5bc54bb0d90bc792ed32cd650cfb87c7de972.